app/code/Magento/Quote/Model/Quote.php

```diff
@@ -2289,7 +2298,11 @@ public function validateMinimumAmount($multishipping = false)
             $baseTotal = 0;
             foreach ($addresses as $address) {
                 $taxes = ($taxInclude) ? $address->getBaseTaxAmount() : 0;
-                $baseTotal += $address->getBaseSubtotalWithDiscount() + $taxes;
+                if ($includeDiscount) {
```

Comment:
The current implementation of `getBaseSubtotalWithDiscount` method is as simple as `$this->getBaseSubtotal() + $this->getBaseDiscountAmount()`. If there is no bugs related to tax in current implementation I would suggest simply returning `$this->getBaseSubtotal() + $taxes` or `$address->getBaseSubtotalWithDiscount() + $taxes` depending on `includeDiscount` configuration value.

Please consider the same for other instances.

Reply:
Hi @sivaschenko,
changes implemented